Zend/zend_exceptions.stub.php

```diff
@@ -24,14 +24,11 @@ class Exception implements Throwable
 {
     /** @var string */
     protected $message = "";
-    /** @var string */
-    private $string = "";
+    private string $string = "";
     /** @var int */
     protected $code = 0;
-    /** @var string|null */
-    protected $file = null;
-    /** @var int|null */
-    protected $line = null;
+    protected string $file = "";
```

Comment:
Why did these defaults switch from null to ""/0?

Reply:
As far as I saw, null values are always converted "" or 0, and I thought that it would be better to make these properties non-nullable instead. Would it be problematic BC-wise?

Reply:
Ah yes, the getFile() and getLine() methods aren't nullable, so it doesn't really make sense to make the properties nullable. I think from a modelling perspective using null would have been better to indicate "not available", but clearly that ship has sailed.